controller/src/main/java/org/jboss/as/controller/SimpleResourceDefinition.java

```diff
@@ -102,6 +104,7 @@ public SimpleResourceDefinition(final PathElement pathElement, final Description
         this.minOccurs = 0;
         this.maxOccurs = Integer.MAX_VALUE;
         this.feature = true;
+        this.additionalPackages = null;
```

Comment:
maybe initialize it as an empty array so that the `null`check is not necessary?

Reply:
BTW, this field can be initialised from Parameters.additionalPackages  that can be null too. 
Not calling into API if the field is null seems fine to me (and common approach in this class).